web/models/attendance/attendance_helper.ex

```diff
@@ -6,22 +6,36 @@ defmodule CoursePlanner.AttendanceHelper do
 
   alias CoursePlanner.{Repo, OfferedCourse}
 
+  def get_course_attendances(offered_course_id) do
+    List.first(Repo.all(from oc in OfferedCourse,
+      join: s in assoc(oc, :students),
+      join: c in assoc(oc, :classes),
+      join: a in assoc(c,  :attendances),
+      preload: [:term, :course, :teachers, students: s],
+      preload: [classes: {c, attendances: a}],
+      where: oc.id == ^offered_course_id and is_nil(s.deleted_at),
+      order_by: [asc: c.date]))
+  end
+
   def get_all_offered_courses do
-    Repo.all(from OfferedCourse,
-      preload: [:term, :course, :teachers])
+    Repo.all(from oc in OfferedCourse,
+      join: c in assoc(oc, :classes),
```

Comment:
when I do not add this join then offered courses, it returns offered_courses without classes too. this was the main intention behind adding this join. So if there is no class yet added for an offered_course then it won't be returned